bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastBridgeHandler.java

```diff
@@ -0,0 +1,182 @@
+
+    private @Nullable ScheduledFuture<?> listenerJob;
+    private final UdpListener udpListener;
+    //////////////////
+    // private final ScheduledExecutorService scheduler = Executors
+    // .newSingleThreadExecutor(new NamedThreadFactory(thing.getUID().getAsString(), true));
+    private @Nullable ExecutorService executor;
+    private @Nullable Future<?> eventListenerJob;
+    private static final int UDP_PORT = 41100;
+    private static final int SOCKET_TIMEOUT_MILLISECONDS = 3000;
+    private static final int BUFFER_SIZE = 5120;
+    private @Nullable DatagramSocket socket;
+    private Runnable eventListenerRunnable = () -> {
+        receivePackets();
+    };
+
+    private void receivePackets() {
+        try {
+            DatagramSocket s = new DatagramSocket(null);
+            s.setSoTimeout(SOCKET_TIMEOUT_MILLISECONDS);
+            s.setReuseAddress(true);
+            InetSocketAddress address = new InetSocketAddress(UDP_PORT);
+            s.bind(address);
+            socket = s;
+            logger.debug("YXC - UDP Listener got socket on port {} with timeout {}", UDP_PORT,
+                    SOCKET_TIMEOUT_MILLISECONDS);
+        } catch (SocketException e) {
+            logger.debug("YXC - UDP Listener got SocketException: {}", e.getMessage(), e);
+            socket = null;
+            return;
+        }
+
+        DatagramPacket packet = new DatagramPacket(new byte[BUFFER_SIZE], BUFFER_SIZE);
+        while (socket != null) {
+            try {
+                socket.receive(packet);
+                String received = new String(packet.getData(), 0, packet.getLength());
+                String trackingID = UUID.randomUUID().toString().replace("-", "").substring(0, 32);
+                logger.debug("YXC - Received packet: {} (Tracking: {})", received, trackingID);
+                handleUDPEvent(received, trackingID);
+            } catch (SocketTimeoutException e) {
+                // Nothing to do on socket timeout
+            } catch (IOException e) {
+                logger.debug("YXC - UDP Listener got IOException waiting for datagram: {}", e.getMessage());
+                socket = null;
+            }
+        }
+        logger.debug("YXC - UDP Listener exiting");
+    }
+
+    /////////////////////////
+    public YamahaMusiccastBridgeHandler(Bridge bridge) {
+        super(bridge);
+        udpListener = new UdpListener(this);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.ONLINE);
+        startUDPListenerJob();
+        /////////
+        if (eventListenerJob == null || eventListenerJob.isCancelled()) {
+            executor = Executors.newSingleThreadExecutor(new NamedThreadFactory("binding-yamahamusiccast"));
+            eventListenerJob = executor.submit(eventListenerRunnable);
```

Comment:
The Runnable can be replaced by this lambda:
```suggestion
            eventListenerJob = executor.submit(this::receivePackets);
```

Reply:
Done, will be available with next push when code is cleaned up.